server/text_generation_server/models/flash_causal_lm.py

```diff
@@ -579,49 +708,41 @@ def generate_token(
             # Prefill
             if prefill:
                 # Remove generated token to only have prefill and add nan for first prompt token
-                prefill_logprobs = [float("nan")] + logprobs.gather(
-                    1, all_input_ids_tensor[1:input_length].unsqueeze(1)
-                ).squeeze(1)[:-1].tolist()
+                request_prefill_logprobs = [float("nan")] + prefill_logprobs[
+                    start_index : end_index - 1
+                ]
                 prefill_token_ids = all_input_ids[:-1]
                 prefill_texts = self.tokenizer.batch_decode(
                     prefill_token_ids,
```

Comment:
`.cpu()` here for a big speedup

Reply:
What?

Reply:
Ah, that's different from `CausalLM` where `prefill_token_ids` is a tensor.